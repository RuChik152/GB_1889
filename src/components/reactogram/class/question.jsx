import React, { Component, useRef } from "react";

export class Question extends Component {



    render() {
        return <div className = "modal">
            <p>Привет, прежде чем приступить к работе, укажи свое имя</p>
            <input type = "text" value = { this.props.author } onChange = { this.props.change } className = "modal__name"/>
            <button onClick = { this.props.quest }>Отправить</button>
        </div>
    }
}