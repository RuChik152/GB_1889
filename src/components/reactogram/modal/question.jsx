import React, { Component } from 'react';
//import './question.scss';

export class Question extends Component {
  render() {
    return (
      <div className="modal">
        <div className="modal__item">
          <p className="modal__text">
            Привет, прежде чем приступить к работе, укажи свое имя
          </p>
          <input
            type="text"
            value={this.props.author}
            onChange={this.props.change}
            className="modal__name"
            placeholder="Ваше Имя"
          />
          <button
            className="modal__btn"
            onClick={this.props.quest}
            data-testid="quest-data-test"
          >
            Отправить
          </button>
        </div>
      </div>
    );
  }
}
