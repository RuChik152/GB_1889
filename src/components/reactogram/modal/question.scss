.modal {
  width: 1140px;
  margin: 44vh auto;
}

.modal__item {
  display: flex;
  flex-direction: column;
  text-align: center;
  font-size: 3vw;
  align-items: center;
}

.modal__name {
  width: 28%;
  height: 6vh;
  outline: none;
  border: none;
  padding: 8px;
  font-size: 22px;
  color: #d94d67;
  border-radius: 4px;
  background-color: #ffffffd1;
  transition: 0.4s ease-in-out;
  &:focus {
    box-shadow: 0px 0px 16px 10px rgba(0,0,0, 0.14);
    background-color: white;
  }
}

.modal__btn {
  font-size: 2vw;
  padding: 7px;
  margin-top: 13px;
  border: none;
  border-radius: 4px;
  color: #5e6c22;
  background-color: #ffffffd1;
  cursor: pointer;
  &:hover {
    background-color: white;
  }
}

.modal__text{
  width: 53vw;
  display: inline-block;
  margin-bottom: 17px;
  color: #583b0f;
}

