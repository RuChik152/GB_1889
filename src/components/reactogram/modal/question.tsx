import React, { FC } from 'react';
import style from './question.module.scss';

interface QuestionProps {
    author: boolean,
    quest: () => void,
    change: () => void
}

export const Question: FC<QuestionProps> = ( { author, change, quest }) => {
    return (
        <div className={style.modal}>
          <div className={style.modal__item}>
            <p className={style.modal__text}>
              Привет, прежде чем приступить к работе, укажи свое имя
            </p>
            <input
                type="text"
                value={author}
                onChange={change}
                className={style.modal__name}
                placeholder="Ваше Имя"
            />
            <button
                className={style.modal__btn}
                onClick={quest}
                data-testid="quest-data-test"
            >
              Отправить
            </button>
          </div>
        </div>
    );

}
