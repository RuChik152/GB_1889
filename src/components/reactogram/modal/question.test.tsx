import React from 'react';
import { render, waitFor,  screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Question } from './question';
import userEvent from '@testing-library/user-event';

describe('Question', () => {
  it('Render', () => {
    render(<Question author={'test_name'} quest={jest.fn()} change={jest.fn()} />);
  });

  it('Find input field and button', () => {
    const { getByPlaceholderText, getByRole } = render(<Question author={'test_name'} quest={jest.fn()} change={jest.fn()} />);
    const inputText = getByPlaceholderText(/Ваше Имя/i);
    const btn = getByRole('button');
    expect(inputText).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
  });

  it('Text in INPUT', () => {
    render(<Question author={'test_name'} quest={jest.fn()} change={jest.fn()} />);
    const link = screen.getByPlaceholderText(/Ваше Имя/i);
    expect(link).toHaveValue('test_name');
  });

  it('Click to button', async () => {
    const handelClick = jest.fn();
    const { getByRole } = render(<Question quest={handelClick} author={'test_name'} change={jest.fn()}/>);
    const btn = getByRole('button');
    userEvent.click(btn);
    await waitFor(() => expect(handelClick).toHaveBeenCalledTimes(1), {
      timeout: 1000,
    });
  });

  it('Snapshot', () => {
    const { asFragment } = render(<Question author={'test_name'} quest={jest.fn()} change={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
