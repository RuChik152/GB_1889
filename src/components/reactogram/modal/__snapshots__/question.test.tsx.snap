// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal__item"
    >
      <p
        class="modal__text"
      >
        Привет, прежде чем приступить к работе, укажи свое имя
      </p>
      <input
        class="modal__name"
        id="input_test_id"
        placeholder="Ваше Имя"
        type="text"
        value="test_name"
      />
      <button
        class="modal__btn"
        data-testid="quest-data-test"
      >
        Отправить
      </button>
    </div>
  </div>
</DocumentFragment>
`;
